File: vortexdbg-android/src/main/java/com/vortexdbg/linux/android/dvm/wrapper/DvmInteger.java
package com.vortexdbg.linux.android.dvm.wrapper;

import com.vortexdbg.linux.android.dvm.DvmObject;
import com.vortexdbg.linux.android.dvm.VM;

public class DvmInteger extends DvmObject<Integer> {

    @SuppressWarnings("unused")
    public static DvmInteger valueOf(VM vm, int i) {
        return new DvmInteger(vm, i);
    }

    private DvmInteger(VM vm, Integer value) {
        super(vm.resolveClass("java/lang/Integer"), value);
    }
}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/android/dvm/wrapper/DvmInteger.kt
package com.vortexdbg.linux.android.dvm.wrapper

import com.vortexdbg.linux.android.dvm.DvmObject
import com.vortexdbg.linux.android.dvm.VM

class DvmInteger private constructor(vm: VM, value: Int) : DvmObject<Int>(vm.resolveClass("java/lang/Integer"), value) {

    companion object {
        @JvmStatic
        @Suppress("unused")
        fun valueOf(vm: VM, i: Int): DvmInteger {
            return DvmInteger(vm, i)
        }
    }
}
